packages/bi/src/index.ts

```diff
@@ -0,0 +1,175 @@
+    return toBI(JSBI.multiply(this.jsbi, toJSBI(other)));
+  }
+
+  mod(other: BIish): BI {
+    return toBI(JSBI.remainder(this.jsbi, toJSBI(other)));
+  }
+
+  abs(): BI {
+    if (JSBI.greaterThanOrEqual(this.jsbi, toJSBI(0))) {
+      return toBI(this.jsbi);
+    } else {
+      return toBI(JSBI.unaryMinus(this.jsbi));
+    }
+  }
+
+  pow(other: BIish): BI {
+    return toBI(JSBI.exponentiate(this.jsbi, toJSBI(other)));
+  }
+
+  and(other: BIish): BI {
+    return toBI(JSBI.bitwiseAnd(this.jsbi, toJSBI(other)));
+  }
+
+  or(other: BIish): BI {
+    return toBI(JSBI.bitwiseOr(this.jsbi, toJSBI(other)));
+  }
+
+  xor(other: BIish): BI {
+    return toBI(JSBI.bitwiseXor(this.jsbi, toJSBI(other)));
+  }
+
+  not(): BI {
+    return toBI(JSBI.bitwiseNot(this.jsbi));
+  }
+
+  mask(other: BIish): BI {
+    const jsbiOther = toJSBI(other);
+    if (
+      JSBI.lessThan(jsbiOther, toJSBI(0)) ||
+      JSBI.lessThan(this.jsbi, toJSBI(0))
+    ) {
+      throw new Error("mask works only with positive numbers");
+    }
+
+    const length = toJSBI(this.jsbi.toString(2).length);
+    if (JSBI.lessThanOrEqual(length, jsbiOther)) {
+      return toBI(this.jsbi);
+    } else {
+      const maskNum = JSBI.leftShift(
+        JSBI.signedRightShift(this.jsbi, jsbiOther),
+        jsbiOther
+      );
+      return toBI(JSBI.bitwiseXor(this.jsbi, maskNum));
+    }
+  }
+
+  shl(other: BIish): BI {
+    return toBI(JSBI.leftShift(this.jsbi, toJSBI(other)));
+  }
+
+  shr(other: BIish): BI {
+    return toBI(JSBI.signedRightShift(this.jsbi, toJSBI(other)));
+  }
+
+  eq(other: BIish): boolean {
+    return JSBI.equal(this.jsbi, toJSBI(other));
+  }
+
+  lt(other: BIish): boolean {
+    return JSBI.lessThan(this.jsbi, toJSBI(other));
+  }
+
+  lte(other: BIish): boolean {
+    return JSBI.lessThanOrEqual(this.jsbi, toJSBI(other));
+  }
+
+  gt(other: BIish): boolean {
+    return JSBI.greaterThan(this.jsbi, toJSBI(other));
+  }
+
+  gte(other: BIish): boolean {
+    return JSBI.greaterThanOrEqual(this.jsbi, toJSBI(other));
+  }
+
+  isNegative(): boolean {
+    return JSBI.lessThan(this.jsbi, toJSBI(0));
+  }
+
+  isZero(): boolean {
+    return JSBI.equal(this.jsbi, toJSBI(0));
+  }
+
+  toNumber(): number {
+    return JSBI.toNumber(this.jsbi);
+  }
+
+  toBigInt(): bigint {
+    try {
+      return BigInt(this.jsbi.toString(10));
+    } catch (e) {
+      throw new Error("this platform does not support BigInt");
+    }
+  }
+
+  toString(radix?: number): string {
+    radix = radix || 10;
+    return this.jsbi.toString(radix);
+  }
+
+  toHexString(): string {
+    return this.jsbi.toString(16);
+  }
+
+  static from(value: any): BI {
+    if (value instanceof BI) {
+      return value;
+    } else if (isBIish(value)) {
+      return toBI(toJSBI(value));
+    } else {
+      throw new Error(`invalid type: ${value} can't be converted into BI`);
+    }
+  }
+
+  static isBI(value: any): value is BI {
+    return value instanceof BI;
```

Comment:
it would be broken when a user uses `bi@0.0.1` and `bi@0.0.2`togther.

```ts
import { f } from 'other-lib'
import {BI} from 'bi' // @0.0.2

const num = f(); /* return a BI@0.0.1 */
BI.isBI(num) // false
```
maybe we should [impl](https://github.com/ethers-io/ethers.js/blob/master/packages/bignumber/src.ts/bignumber.ts#L294) `isBI` in this way